File: Data Structures & Algorithms/two-integer-sum/submission-3.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, List<int>> ints = new Dictionary<int, List<int>>();
        for (int i = 0; i < nums.Length; i ++){
            if (!ints.ContainsKey(nums[i]))
                ints[nums[i]] = new List<int>();
            ints[nums[i]].Add(i);
        }

        for (int i = 0; i < nums.Length; i ++){
            if (ints.ContainsKey(target - nums[i])){
                if (ints[target - nums[i]].Count == 2)
                    return new int[] {i, ints[target - nums[i]][1]};
                else if (target - nums[i] != nums[i])
                    return new int[] {i, ints[target - nums[i]][0]};
            }
        }

        return null;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1

        while l < r:
            mid = (l + r) // 2

            if nums[mid] > nums[r]:
                l = mid + 1

            else:
                r = mid

            print(l, r)

        final = nums[r] if nums[l] > nums[r] else nums[l]
        return final

            